agent/engine/dockerclient/versionsupport_windows.go

```diff
@@ -46,9 +46,18 @@ func getWindowsReplaceableVersions() []DockerVersion {
 	}
 }
 
-// getAgentVersions for Windows should return all of the replaceable versions plus additional versions
+// getWindowsSupportedVersions returns the set of remote api versions that are
+// supported by agent in windows
+func getWindowsSupportedVersions() []DockerVersion {
+	return []DockerVersion{
+		Version_1_24,
+		Version_1_30,
```

Comment:
Is there any reason we don't have the versions between 1.24 and 1.30?

Reply:
We add versions here when those api versions' features are used in ECS. Hence, only 1.24 and 1.30 are added here.

Reply:
I haven't read the surrounding code recently enough to know whether there has been a change here I'm forgetting, but...

The initial intent of reporting API versions up to ECS was twofold: First, to enable capability-based placement logic such that we only place a task on an instance that is capable of implementing all the requested configuration options.  Second, it was to enable us to add additional configuration options to the task definition over time and not require updates in the ECS agent.  This second bit is accomplished through accepting a serialized `Config` and `HostConfig` from the ECS backend and applying it to our container creation calls through the Docker API.  We signal all of the available API versions (regardless of whether or not they're used) so that we have this flexibility going forward; we may make use of any of them in the future without requiring further changes to the agent.

Reply:
Got it. If thats the case, we should probably add all of the versions between 1.25 and 1.30 [here](https://github.com/aws/amazon-ecs-agent/blob/dev/agent/engine/dockerclient/versionsupport_unix.go#L32) for Linux too.